onnxruntime/core/providers/ngraph/ngraph_custom_op.cc

```diff
@@ -141,11 +163,10 @@ void NGRAPHCustomOp::Initialize(const OrtCustomOpApi* api, OrtKernelContext* con
 Status NGRAPHCustomOp::Compute(const OrtCustomOpApi* api, OrtKernelContext* context) const {
   Ort::CustomOpApi ort{*api};
 
-  //TODO: Minimize locked region
-  std::lock_guard<std::mutex> lock(compute_lock_);
-
   // Initialize nGraph function if it is not already initialized.
-  Initialize(api, context);
+  if (ng_curr_exe_ == nullptr) {
```

Comment:
I don't think this will work. I think the code assumes that it always has to go through Initialize() to determine compatibility with the input shape.
i.e. even if ng_curr_exe_ is not nullptr, it may not work with the current inputs, whose shape may have changed from the previous inputs.

Reply:
Since this problem didn't show up for me locally I reverted it for now. In the meantime I'll see if it would work and be worth it to check the input shapes too.